Automattic/wp-calypso
# Removing congrats screen from plan purchases

Fixes;
- https://github.com/Automattic/dotcom-forge/issues/9183

Related to;
- https://github.com/Automattic/wp-calypso/pull/94766 
- https://github.com/Automattic/wp-calypso/pull/94929

## Proposed Changes

* Following a similar pattern as in https://github.com/Automattic/wp-calypso/pull/94766
* When only plan purchases are present in the cart, take the user directly to /plans.

<details><summary> 📝 TODO from @lsl </summary>
<p>

- Add tests for thank you function changes
- Confirm whether this should apply to onboarding flows
- Confirm if this should apply to domain + plan purchases, I believe so.
- I'm proposing a thank you notice component in this PR, need to decide if we should keep it or not.

</p>
</details>

## Testing Instructions

Plan upgrades:
* Purchase a plan upgrade from /plans
* You should no longer see the thank you page; instead, go directly to the plans screen.
* You should see a toast notification temporarily.

No changes:
1. Purchases including non-plan products should continue to show a thank you page
1. Plan purchases during sign-up/onboarding should continue to show ~~a thank you page~~ the Goals step
1. eCommerce plan purchase, as there is a special logic for that: https://github.com/Automattic/wp-calypso/blob/f4188f019a5de25a2f0f443a12b7602a69ba7d96/client/my-sites/checkout/get-thank-you-page-url/index.ts#L452-L473
1. Jetpack plan purchases: https://github.com/Automattic/wp-calypso/blob/f4188f019a5de25a2f0f443a12b7602a69ba7d96/client/my-sites/checkout/get-thank-you-page-url/index.ts#L545-L572
